driver-async/src/main/com/mongodb/async/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -144,6 +144,48 @@ public void onResult(final Integer result, final Throwable t) {
         });
     }
 
+    @Override
+    public void skip(final long bytesToSkip, final SingleResultCallback<Long> callback) {
+        notNull("callback", callback);
+        final SingleResultCallback<Long> errHandlingCallback = errorHandlingCallback(callback, LOGGER);
+        if (checkClosed()) {
+            errHandlingCallback.onResult(null, null);
+        } else if (!hasFileInfo()) {
+            getGridFSFile(new SingleResultCallback<GridFSFile>() {
```

Comment:
Because we need a file we have to use a callback.

Reply:
Ack